MDANSE/Src/MDANSE/Framework/Jobs/CurrentCorrelationFunction.py

```diff
@@ -438,7 +440,7 @@ def run_step(self, index: int):
                     curr = np.einsum(
                         "ik,ij->ikj",
                         veloc,
-                        np.exp(1j * temp_dotprod),
+                        np.exp(1j * temp_dotprod) * np.sqrt(qvec_weights[None, :]),
                     )
```

Comment:
Here also, no need for new axis could also do this.

```suggestion
                    curr = np.einsum(
                        "ik,ij,j->ikj",
                        veloc,
                        np.exp(1j * temp_dotprod),
                        np.sqrt(qvec_weights),
                    )
```